L1Trigger/VertexFinder/interface/AnalysisSettings.h

```diff
@@ -0,0 +1,95 @@
+#ifndef __L1Trigger_VertexFinder_AnalysisSettings_h__
+#define __L1Trigger_VertexFinder_AnalysisSettings_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "L1Trigger/VertexFinder/interface/AlgoSettings.h"
+#include "SimTracker/Common/interface/TrackingParticleSelector.h"
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+
+namespace l1tVertexFinder {
+
+  class AnalysisSettings : public AlgoSettings {
+  public:
+    AnalysisSettings(const edm::ParameterSet& iConfig);
+    ~AnalysisSettings() {}
```

Comment:
Can there ever be a case that an `AnalysisSetting` object would be deleted via `AlgoSettings` pointer? I.e. effectively
```cpp
AlgoSettings *s = new AnalysisSettings();
delete s;
```

Reply:
This use case was never envisioned by me, though I didn't write the original code. It's certainly not a use case I would see as useful, efficient, or recommended.

Reply:
Thanks, then this is ok. It's just that if there is a (even remote) possibility that an ownership to an object would be moved through a base class pointer, the destructor of the base class must be declared as `virtual`.